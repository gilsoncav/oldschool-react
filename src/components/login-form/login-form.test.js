import LoginForm from './login-form';

describe('LoginForm component', () => {
  const spyValidate = sinon.spy();
  const spyHistPush = sinon.spy();

  const historyMock = { push: spyHistPush };

  it('renders correctly', () => {
    const testForm = {
      getFieldDecorator: jest.fn(opts => c => c),
      validateFields: jest.fn(),
    };
    const wrapper = shallow(<LoginForm.WrappedComponent history={historyMock} form={testForm} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({ userLogged: true });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleSubmit and simulate error on login, then show message', async (done) => {
    const testForm = {
      getFieldDecorator: jest.fn(opts => jest.fn()),
      validateFields: jest.fn((opts) => {
        false, [{ field1: 'Teste' }];
      }),
    };
    const wrapper = shallow(<LoginForm.WrappedComponent history={historyMock} form={testForm} />);
    await wrapper.instance().handleSubmit(new Event('submit'));
    console.log('Waiting to finish function', wrapper.instance().props.form);
    expect(spyHistPush.callCount).toBe(true);
    expect(spyHistPush.args[0][0]).toBe('/dashboard');
    spyValidate.restore();
    spyHistPush.restore();
    done();
  });
});
