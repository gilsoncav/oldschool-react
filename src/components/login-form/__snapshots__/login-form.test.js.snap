// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders correctly 1`] = `
<Form
  className="login-form"
  colon={true}
  hideRequiredMark={false}
  labelAlign="left"
  labelCol={
    Object {
      "offset": 2,
      "span": 20,
    }
  }
  layout="horizontal"
  onSubmit={[Function]}
  wrapperCol={
    Object {
      "offset": 2,
      "span": 20,
    }
  }
>
  <FormItem
    hasFeedback={false}
  >
    <Input
      autoComplete="username"
      disabled={false}
      placeholder="Username"
      prefix={
        <Icon
          type="user"
        />
      }
      type="text"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      autoComplete="current-password"
      disabled={false}
      placeholder="Password"
      prefix={
        <Icon
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Checkbox
      className="text-primary"
      indeterminate={false}
    >
      Remember me
    </Checkbox>
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      style={
        Object {
          "width": "100%",
        }
      }
      type="primary"
    >
      Log in
    </Button>
    <Link
      to="/register"
    >
      Not registered? Click here!
    </Link>
  </FormItem>
</Form>
`;

exports[`LoginForm component renders correctly 2`] = `
<Row
  gutter={0}
>
  <Col
    className="text-center"
    offset={2}
    span={22}
  >
    You are already logged in!
  </Col>
  <Col
    offset={2}
    span={20}
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      type="primary"
    >
      Click here to enter.
    </Button>
  </Col>
</Row>
`;
